rollouts/controllers/remoterootsync_controller.go

```diff
@@ -63,7 +63,8 @@ var (
 )
 
 const (
-	externalSyncCreatedConditionType = "ExternalSyncCreated"
+	externalSyncCreatedConditionType  = "ExternalSyncCreated"
```

Comment:
What is the difference between these? It would assume that it being up-to-date would imply that it is also created?

Also, have we thought if we can adopt some of the more common condition types as described in https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus. This might also be relevant: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/308

Reply:
> What is the difference between these? It would assume that it being up-to-date would imply that it is also created?

`ExternalSyncCreated` indicates the sync was created on the target cluster and `ExternalSyncUpToDate` indicates the sync is up to date with the latest configuration of the remote root sync. This does similar, but the difference is after you create a rollout resource to rollout v1 of a package and the rollout completes, the rollout resource is updated to rollout the new v2 version of the package, and if, for whatever reason, the reconciler is unable to update the sync on the target cluster (say the cluster is on-premise and there are temporary connectivity issues), the `ExternalSyncCreated` will remain true since the external sync was created and `ExternalSyncUpToDate` would be set to a non-True value since the connection cannot be made to the external sync to update it from v1 of the package to v2.

> Also, have we thought if we can adopt some of the more common condition types as described in https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus. This might also be relevant: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/pull/308

Looking into this. My initial thought is to avoid the Ready condition since we don't have the correct information yet to determine if all sub-resources (including any resources that the sync creates) is in ready status. Similarly with Reconciling. The Stalled condition I am thinking can be a good candidate. I'll likely reach out to discuss more.